pkg/controllers/machine_controller.go

```diff
@@ -456,17 +456,18 @@ func (r *MachineReconciler) reconcileMachine(ctx context.Context, id string) err
 
 	log.V(1).Info("Reconciling domain")
 	state, volumeStates, nicStates, err := r.reconcileDomain(ctx, log, machine)
+
 	if err != nil {
+		locErr := r.updateAPIMachineStatus(ctx, machine, state, volumeStates, nicStates)
+		if locErr != nil {
+			log.Error(locErr, "failed to update API machine")
+		}
 		return providerimage.IgnoreImagePulling(err)
 	}
```

Comment:
if updateAPIMachineStatus fails and err is ImagePulling error, it will reconcile based on trigger from providerimage, which will be thrown when image will available (of course you don't have current status but machine domain doesn't exist in this time). It is same logic as before. If i return error base on updateAPIMachineStatus with ImagePulling error, we will end in loop, where we will wait for download of image. If err is different as ImagePulling, machine will reconcile again and status can be updated in next reconcile loop.

What do you suggest do here?

Reply:
i kept it as it is.